fusion/dsl/qparser.h

```diff
@@ -0,0 +1,207 @@
+    std::string srcFile;
+
+    //! \brief symbol type
+    //!
+    //! FIXME: to be an enum
+    std::string type;
+};
+
+// ----------------------------------------------------------------
+//! \brief FSL symbol table type
+// ----------------------------------------------------------------
+struct SymbolTable
+{
+    //! \brief symbol table iterator
+    using TableItrType = std::unordered_map<std::string, FslSymbol>::iterator;
+
+    //! \brief look up name in table
+    bool findSymbol(std::string name)
+    {
+        if (table.find(name) != table.end())
+            return true;
+        return false;
+    }
+
+    //! \brief if not already in symtab, insert it
+    void insertSymbol(std::string name, FslSymbol s)
+    {
+        if (!findSymbol(name))
+            table.insert(make_pair(name, s));
+    }
+
+    //! \brief ...
+    void setType(std::string name, std::string type)
+    {
+        TableItrType itr = table.find(name);
+        if (itr != table.end())
+        {
+            itr->second.type = type;
+        }
+    }
+
+    //! \brief ...
+    void clear() { table.clear(); }
+
+    //! \brief symbol info to stream
+    void info(std::ostream & os, bool shortPath = false);
+
+    //! \brief the symbol table
+    std::unordered_map<std::string, FslSymbol> table;
+};
+
+// ----------------------------------------------------------------
+//! \brief QParser support for flex/bison
+//!
+//! This is an edited version of QParser. It supports the fusion
+//! dsl proposal, other functionality was removed, so the structure
+//! looks like over kill. 
+// ----------------------------------------------------------------
+struct QParser
+{
+    //! \brief ...
+    QParser();
+
+    //! \brief ...
+    ~QParser() {}
+
+    //! \brief initialize parser state between code bases
+    void coldReset()
+    {
+        warmReset();
+        reqId = 0;
+        optId = 0;
+        symtab.clear();
+    }
+
+    //! \brief initialize parser state between files
+    void warmReset()
+    {
+        lineNo = 1;
+        curCol = 1;
+        currentFile = "";
+        errMsg = "";
+    }
+
+    //! \brief set the inputFile vector when isInLine
+    void setInputFiles(const std::vector<std::string> & fv) { inputFiles = fv; }
+
+    //! \brief parse all input files
+    bool parse() { return parseFiles(); }
+
+    //! \brief parse all input files
+    bool parseFiles();
+
+    //! \brief parse one file
+    bool parse(std::string);
+
+    //! \brief remove the Msg dependency for QParser
+    void emsg(std::string m) { std::cout << "-E:QP: " << m << std::endl; }
+
+    // --------------------------------------------------------------
+    // Symbol table shim
+    // --------------------------------------------------------------
+    //! \brief look up string in symbol table
+    bool findSymbol(const std::string name) { return symtab.findSymbol(name); }
+
+    //! \brief look up const/char in symbol table
+    bool findSymbol(const char* name)
```

Comment:
Methods can be `const`

Also, these methods don't really find the symbol -- they check existence.  Suggest `hasSymbol`?